DOCS-3556: Increase timeout for sessions (#4061)
viamrobotics/docs: docs/dev/reference/apis/sessions.md

A disconnected client will attempt to establish a new session immediately prior to the next operation it performs.

#### Change the session timeout

> The default session length is 20 seconds.

Reviewer: ```suggestion
The default session timeout length is 20 seconds.
```
technically incorrect otherwise

> To change the session timeout length, pass the `timeout` parameter to the `DialOptions` object:

Reviewer: ```suggestion
To change this, pass the `timeout` parameter to the `DialOptions` object:
```
[opt] for brevity; no need to repeat esp. since `timeout` is a somewhat descriptive parameter name

{{< tabs >}}
{{% tab name="Python" %}}

```python {class="line-numbers linkable-line-numbers" data-line="1"}
opts = RobotClient.Options(dial_options=DialOptions(timeout=10)).with_api_key(

>   # Replace "<API-KEY>" (including brackets) with your machine's api key
>   api_key='<API-KEY>',
>   # Replace "<API-KEY-ID>" (including brackets) with your machine's api key id

Reviewer: ```suggestion
  # Replace "<API-KEY>" (including brackets) with your machine's API key
  api_key='<API-KEY>',
  # Replace "<API-KEY-ID>" (including brackets) with your machine's API key ID
```
[nit/opt] For all of these, correct caps would be nice but I know app doesn't generate it that way

Author: fixed in app as well.

  api_key_id='<API-KEY-ID>'
)
await RobotClient.at_address('<machine address>', opts)
```

{{% /tab %}}
{{% tab name="Go" %}}

> ```go {class="line-numbers linkable-line-numbers" data-line="4,10,14"}

Reviewer: ```suggestion
```go {class="line-numbers linkable-line-numbers" data-line="4,11,15"}
```

// Import the time package in addition to the other imports:
import (
  ...
  "time"
  ...
)

// Configure a timeoutContext, then pass this context to the client:
func main() {
  ctx := context.Background()
  timeoutContext, cancel := context.WithTimeout(ctx, 10*time.Second)
  logger := logging.NewDebugLogger("client")

  machine, err := client.New(
    timeoutContext,
    "<machine address>",
    logger,
    client.WithDialOptions(rpc.WithEntityCredentials(
      /* Replace "<API-KEY-ID>" (including brackets) with your machine's

>         api key id */

Author: ```suggestion
        API key id */
```

      "<API-KEY-ID>",
      rpc.Credentials{
        Type:    rpc.CredentialsTypeAPIKey,

>         /* Replace "<API-KEY>" (including brackets) with your machine's api key */

Author: ```suggestion
        /* Replace "<API-KEY>" (including brackets) with your machine's API key */
```

        Payload: "<API-KEY>",
      })),
  )
  ...
}
```

{{% /tab %}}
{{% tab name="TypeScript" %}}

```ts {class="line-numbers linkable-line-numbers" data-line="11"}
const machine = await VIAM.createRobotClient({
  host,
  credentials: {
    type: "api-key",

>     /* Replace "<API-KEY>" (including brackets) with your machine's api key */
>     payload: "<API-KEY>",
>     authEntity: "<API-KEY-ID>",
>     /* Replace "<API-KEY-ID>" (including brackets) with your machine's api key id */

Author: ```suggestion
    /* Replace "<API-KEY>" (including brackets) with your machine's API key */
    payload: "<API-KEY>",
    authEntity: "<API-KEY-ID>",
    /* Replace "<API-KEY-ID>" (including brackets) with your machine's API key id */
```

  },
  signalingAddress: "https://app.viam.com:443",
  dialTimeout: 1000,
});
```

{{% /tab %}}
{{% tab name="C++" %}}

```cpp {class="line-numbers linkable-line-numbers" data-line="8"}
std::string host("guardian-main.vw3iu72d8n.viam.cloud");
DialOptions dial_opts;
dial_opts.set_entity(std::string("<API-KEY-ID>"));

> /* Replace "<API-KEY-ID>" (including brackets) with your machine's api key id */
> Credentials credentials("api-key", "<API-KEY>");
> /* Replace "<API-KEY>" (including brackets) with your machine's api key */

Author: ```suggestion
/* Replace "<API-KEY-ID>" (including brackets) with your machine's API key id */
Credentials credentials("api-key", "<API-KEY>");
/* Replace "<API-KEY>" (including brackets) with your machine's API key */
```

> /* Replace "<API-KEY-ID>" (including brackets) with your machine's api key id */
> Credentials credentials("api-key", "<API-KEY>");
> /* Replace "<API-KEY>" (including brackets) with your machine's api key */
> dial_opts.set_credentials(credentials);
> dial_opts.set_timeout(std::chrono::duration<float>(10));
> boost::optional<DialOptions> opts(dial_opts);
> Options options(0, opts);
> 
> auto machine = RobotClient::at_address(host, options);
> ```
> 
> {{% /tab %}}
> {{% tab name="Flutter" %}}
> 
> ```dart {class="line-numbers linkable-line-numbers" data-line="11"}
> Future<void> connectToViam() async {
>   const host = '<machine-address>';
>   /* Replace "<API-KEY-ID>" (including brackets) with your machine's api key id */
>   const apiKeyID = '<API-KEY-ID>';
>   /* Replace "<API-KEY>" (including brackets) with your machine's api key */

Author: ```suggestion
/* Replace "<API-KEY-ID>" (including brackets) with your machine's API key id */
Credentials credentials("api-key", "<API-KEY>");
/* Replace "<API-KEY>" (including brackets) with your machine's API key */
dial_opts.set_credentials(credentials);
dial_opts.set_timeout(std::chrono::duration<float>(10));
boost::optional<DialOptions> opts(dial_opts);
Options options(0, opts);

auto machine = RobotClient::at_address(host, options);
```

{{% /tab %}}
{{% tab name="Flutter" %}}

```dart {class="line-numbers linkable-line-numbers" data-line="11"}
Future<void> connectToViam() async {
  const host = '<machine-address>';
  /* Replace "<API-KEY-ID>" (including brackets) with your machine's API key id */
  const apiKeyID = '<API-KEY-ID>';
  /* Replace "<API-KEY>" (including brackets) with your machine's API key */
```

>   /* Replace "<API-KEY-ID>" (including brackets) with your machine's api key id */
>   const apiKeyID = '<API-KEY-ID>';
>   /* Replace "<API-KEY>" (including brackets) with your machine's api key */

Author: ```suggestion
  /* Replace "<API-KEY-ID>" (including brackets) with your machine's API key id */
  const apiKeyID = '<API-KEY-ID>';
  /* Replace "<API-KEY>" (including brackets) with your machine's API key */
```
